<!-- https://github.com/ros-controls/ros_controllers/tree/kinetic-devel/diff_drive_controller/test -->
<launch>
    <!--arg name="model" default="$(env auto_rover_MODEL)" doc="model type [auto_rover, remo]"/-->
    <arg name="model" default="auto_rover" doc="model type [auto_rover, remo]"/>
    <arg name="description_package" default="$(eval find(arg('model') + '_description'))" />
    <arg name="urdf_file" default="$(find xacro)/xacro '$(arg description_package)/urdf/$(arg model).urdf.xacro'" />

    <!-- Load the URDF model of auto_rover into the ROS Parameter Server -->
    <param name="auto_rover/robot_description" command="$(arg urdf_file)" />

    <!-- Run the auto_rover_base -->
    <!-- This will create the auto_rover hardware interface and instantiate a new controller manager with it -->
    <node name="auto_rover_base" pkg="auto_rover_base" type="auto_rover_base" ns="auto_rover" output="screen"/>

    <!-- Load controller config to the parameter server -->
    <rosparam command="load" 
              file="$(find auto_rover_control)/config/auto_rover_control.yaml"/>

    <!-- Load base config to the parameter server -->
    <rosparam command="load" 
              file="$(find auto_rover_base)/config/base.yaml"/>

    <!-- Load the controllers -->
    <!-- This will use the controller manager that got started inside the difbot_base node -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" ns="auto_rover" args="joint_state_controller
                                            mobile_base_controller"/>
</launch>